create_commits.py

```diff
@@ -0,0 +1,525 @@
+    ####### Get the list of files changed in the commit.                                ###########
+    ################################################################################################
+
+    # Get a list of files changed in a specific commit. See
+    # https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---name-status for more on the
+    # --name-status parameter.
+    result = subprocess.run(
+        ["git", "diff", "--name-status", f"{commit_hash}^", commit_hash],
+        stdout=subprocess.PIPE,
+        text=True,
+        check=True,
+    )
+    files_changed_by_commit: list[str] = result.stdout.splitlines()
+
+    logging.debug(
+        "Output of git diff --name-status:\n %s",
+        "\n".join(files_changed_by_commit),
+    )
+
+    ################################################################################################
+    ####### Create FileAddition and FileDeletion objects for each file changed in the commit. ######
+    ################################################################################################
+    for file_change_line in files_changed_by_commit:
+        # the --name-status parameter in git diff returns lines in the format "status\tfilenames",
+        # where status is a single letter representing the change type, and filenames is a
+        # tab-separated list of filenames. See
+        # https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203
+        # for more on what each status code means.
+
+        # Since we're only inspecting full commits (not the
+        # contents of the working tree, the index, or the files on disk), we only need to handle the
+        # A, M, R, and D status codes.
+        status, *filenames = file_change_line.split("\t")
+
+        logging.debug("status: %s", status)
+        logging.debug("filenames: %s", filenames)
+
+        ############# Handle Additions and Modifications #############
+        if status in ["A", "M"]:
+            logging.debug("Added or modified file %s detected", filenames[0])
+
+            # Per Github's docs on modeling file changes:
+            # https://docs.github.com/en/graphql/reference/input-objects#modeling-file-changes,
+            # new files and modifications to existing files are modeled identically. Each should be
+            # specified as an 'addition', with the full file contents provided.
+
+            file_changes["additions"].append(
+                FileAddition(
+                    path=filenames[-1],
+                    contents=get_file_contents_at_commit(commit_hash, filenames[0]),
+                )
+            )
+
+        ############# Handle Renames #############
+        elif (
+            # We check for "R" in status because we've seen odd formats for rename status codes,
+            # like "R100"
+            "R" in status
+            or status == "R"
```

Comment:
could this be `status.startswith("R")`?

either way, if `"R" not in status`, then `status != "R"` so the `or` condition is extraneous